File: csharp/04_explain_how_to_use_the_tool/Agent/Agent.Application/Agent.cs
namespace Agent.Application;

public class Agent()
{
    private readonly IUserInput _input;
    private readonly IDisplay _display;
    private readonly ILanguageModel _model;

    public Agent(IUserInput input, ILanguageModel model, IDisplay display) : this()
    {
        _input = input;
        _display = display;
        _model = model;
    }

    public void Run()
    {
        var context = new List<Message>();

        while (true)
        {
            var userInput = _input.GetInput();
            if (string.IsNullOrWhiteSpace(userInput))
            {
                break;
            }
            _display.Show("User: " + userInput);
            context.Add(new Message("user", userInput));
            var answer = _model.Prompt(context);
            context.Add(answer);
            _display.Show("Assistant: " + answer.Content);

        }
    }
}

File: java/01_send_user_input_to_model/agent-tests/src/test/java/com/workshop/agent/application/Agent.java
package com.workshop.agent.application;

public class Agent {
    private final UserInput input;
    private final Display display;

    public Agent(UserInput input, Display display) {
        this.input = input;
        this.display = display;
    }

    public void run() {
        Message userMessage = input.read();
        display.show(userMessage.toString());
    }

}
